pkg/registry/client.go

```diff
@@ -229,8 +229,23 @@ type (
 	}
 )
 
+// Deprecated: will be removed in Helm 4
```

Comment:
We need to log/warn IMHO for this deprecation to easily happen. People won't know to fix their login commands, and they will confusing break for users when updating to Helm 4.

(I've included logging in my suggestion below; not sure if that is the best way)